docs/scripts/make_model_zoo_docs.py

```diff
@@ -0,0 +1,225 @@
+"""
+Script for generating the model zoo docs page contents
+``docs/source/user_guide/model_zoo/models.rst ``.
+
+| Copyright 2017-2020, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+import logging
+import os
+
+from jinja2 import Environment, BaseLoader
+
+import eta.core.utils as etau
+
+import fiftyone.zoo as foz
+
+
+logger = logging.getLogger(__name__)
+
+
+_SECTION_TEMPLATE = """
+.. _model-zoo-{{ link_name }}-models:
+
+{{ header_name }} models
+{{ '-' * (header_name|length + 7) }}
+
+Available models
+________________
+
+.. table::
+    :widths: 40 60
+
+    +{{ '-' * col1_width }}+{{ '-' * col2_width }}+
+    | Model name {{ ' ' * (col1_width - 12) }}| Tags {{ ' ' * (col2_width - 6) }}|
+    +{{ '=' * col1_width }}+{{ '=' * col2_width }}+
+{% for model in models %}
+    | :ref:`{{ model['name'] }} <model-zoo-{{ model['name'] }}>` {{ ' ' * (col1_width - 2 * model['name']|length - 22) }}| {{ model['tags_str'] }} {{ ' ' * (col2_width - model['tags_str']|length - 2) }}|
+    +{{ '-' * col1_width }}+{{ '-' * col2_width }}+
+{% endfor %}
+"""
+
+_MODEL_TEMPLATE = """
+.. _model-zoo-{{ name }}:
+
+{{ header_name }}
+{{ '_' * header_name|length }}
+
+{{ description }}.
+
+**Details**
+
+-   Model name: ``{{ name }}``
+-   Model source: {{ source }}
+-   Model size: {{ size }}
+-   Exposes embeddings? {{ exposes_embeddings }}
+-   Tags: ``{{ tags }}``
+
+**Requirements**
+
+{% if base_packages %}
+-   Packages: ``{{ base_packages }}``
+
+{% endif %}
+-   CPU
+
+    -   Support? {{ supports_cpu }}
```

Comment:
I love the automatically generated docs. The only thing that stands out is that the formatting of this seems a bit awkward. I'd change it to something like 
```
- CPU Support
    - yes
    - tensorflow>2
```
for example

Reply:
Okay I went with

```
- CPU support
    - yes
    - Packages: `tensorflow>2`
```

I think `Packages` should be there for consistency with how it looks when there are compute-independent package requirements:

```
- Packages: `torch, torchvision`
- CPU support
    - yes
```

I don't really like the `yes/no` business at all, but it is also being used for `Exposes embeddings? yes/no`, so we should probably revisit in general in the future.

Reply:
>  It would be cool to separate them into cards like the tutorials so that you can search by individual tags and see an example picture of the outputs on the right side of the card. Then when you click on them it brings you here to the details.

Definitely agree on this point, but didn't have the time to make it happen at the moment. It will require some modifications to the code that we're inheriting from the PyTorch docs for the tutorials cards.

I'll make an issue out of this one.

Reply:
https://github.com/voxel51/fiftyone/issues/742.